### PR TITLE
fix #100437 Improve 'terminal.integrated.commandsToSkipShell' info

### LINKED ISSUE #100437: Settings editor only gives a partial list of the defaults for 'terminal.integrated.commandsToSkipShell'
Issue Type: <b>Feature Request</b>

An easily-overlooked ellipsis is the only clue that the list is only a partial one.

image

The actual list of defaults can be viewed after running `Preferences: Open Default Settings (JSON)` and searching there for the string `SkipShell":`

There are currently 118 default items. It is impractical to list them all in the settings editor, but the description text for the setting should be improved to guide the user to find the full list. For example, #100304 means it is now possible to offer a clickable link that will run the command mentioned above. The user could also be advised what string to search for.

VS Code version: Code 1.46.0 (a5d1cc28bb5da32ec67e86cc50f84c67cc690321, 2020-06-10T09:03:20.462Z)
OS version: Windows_NT x64 10.0.18363

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
Improve 'terminal.integrated.commandsToSkipShell' information in Settings Editor

This PR fixes #100437

New display is:
image

Note the use of two command links to help user view the full list of defaults.

Previously it was:
image

### CODE REVIEW

**Comment by gjsjohnmurray** [comment]
@roblourens how about this?

**Comment by Tyriar** [comment]
I was thinking more pushing the [...] up and making it a link to expand the section fully.

**Comment by gjsjohnmurray** [comment]
> 
> 
> I was thinking more pushing the [...] up and making it a link to expand the section fully.

@Tyriar fine by me, but maybe consider taking some/all of the revisions I've made to the preamble text in this PR?

**Comment by roblourens** [comment]
Yeah I would rather not tell the user to click these links, but I'll leave it to @Tyriar whether he wants the text changes

**Comment by gjsjohnmurray** [comment]
> Yeah I would rather not tell the user to click these links

Me too. If we could specify a `command:` URI that ran multiple commands, and if `workbench.action.openRawDefaultSettings` took a parameter that triggered a search, maybe it could be reduced to a single click.

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts [hunk 1]
```diff
@@ -218,7 +218,7 @@ export const terminalConfiguration: IConfigurationNode = {
...
+			markdownDescription: localize('terminal.integrated.commandsToSkipShell', "A set of command IDs whose keybindings will not be sent to the shell but instead always be handled by VS Code. This allows keybindings that would normally be consumed by the shell to act instead the same as when the terminal is not focused, for example `Ctrl+P` to launch Quick Open.\n\n---\n\nMany commands are skipped by default. To override a default and pass that command's keybinding to the shell instead, add the command prefixed with the `-` character. For example add `-workbench.action.quickOpen` to allow `Ctrl+P` to reach the shell.\n\n*The following list of default skipped commands is truncated when viewed in Settings Editor. To see the full list, click [here](command:workbench.action.keepEditor 'Keep this editor open'), then [here](command:workbench.action.openRawDefaultSettings 'Open Default Settings (JSON)'), then search for the first command from the list below.*\n\n---\n\nDefault Skipped Commands:\n\n{0}", DEFAULT_COMMANDS_TO_SKIP_SHELL.sort().map(command => `- ${command}`).join('\n')),
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:221]: > To see the full list, click [here](command:workbench.action.keepEditor 'Keep this editor open'), then [here](command:workbench.action.openRawDefaultSettings 'Open Default Settings (JSON)'), then search for the first command from the list below.

Using [here] is bad practice for screen readers, I suggest:

> To see the full list, [open the default settings JSON](command:workbench.action.openRawDefaultSettings 'Open Default Settings (JSON)') and search for the first command from the list below.

I removed the first part as 2 steps complicates things a bit.

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts (hunk 1 above)
Comment (gjsjohnmurray) [reply src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts]: @Tyriar thanks for the feedback. I have pushed a change.

**Review by Tyriar — APPROVED** [review]
Thanks and sorry about the delay

**Pull request merged**